sdk/test-utils/test-utils/src/tracing/azureTracing.ts

```diff
@@ -0,0 +1,127 @@
+import { OperationTracingOptions, useInstrumenter } from "@azure/core-tracing";
+import { assert } from "chai";
+import { TestInstrumenter } from "./testInstrumenter";
+import { SpanGraph, SpanGraphNode } from "./spanGraphModel";
+// this is the plugin used in the test file
+function chaiAzureTrace(chai: Chai.ChaiStatic, _utils: Chai.ChaiUtils): void {
+  // expect(() => {}).to.supportsTracing() syntax
+  chai.Assertion.addMethod("supportsTracing", function(
+    expectedSpanNames: string[],
+    options?: any,
+    thisArg?: any
+  ) {
+    return supportsTracing(this._obj, expectedSpanNames, options, thisArg);
+  });
+  // assert.supportsTracing(() => {}) syntax
+  chai.assert.supportsTracing = supportsTracing;
+}
+
+// Implementation for supportsTracing Plugin
+async function supportsTracing<
+  Options extends { tracingOptions?: OperationTracingOptions },
+  Callback extends (options: Options) => unknown
+>(
+  callback: Callback,
+  expectedSpanNames: string[],
+  options?: Options,
+  thisArg?: ThisParameterType<Callback>
+) {
+  const instrumenter = new TestInstrumenter();
+  useInstrumenter(instrumenter);
+  const startSpanOptions = {
+    packageName: "test",
+    ...options
+  };
+  const { span: rootSpan, tracingContext } = instrumenter.startSpan("root", startSpanOptions);
+
+  const newOptions = {
+    ...options,
+    tracingOptions: {
+      tracingContext: tracingContext
+    }
+  } as Options;
+  await callback.call(thisArg, newOptions);
+  console.log(callback.name);
+  // TODO: this expectation should be: assert only 1 _root_ span
+  console.log(instrumenter.startedSpans);
+  const spanGraph = getSpanGraph(rootSpan.spanContext.traceId, instrumenter);
+  console.log("----SPAN GRAPH----");
+  console.log(JSON.stringify(spanGraph, null, 2));
+  assert.equal(spanGraph.roots.length, 1, "There should be just one root span");
+  assert.equal(spanGraph.roots[0].name, "root");
+  assert.strictEqual(
+    rootSpan,
+    instrumenter.startedSpans[0],
+    "The root span should match what was passed in."
+  );
+
+  const directChildren = spanGraph.roots[0].children.map((child) => child.name);
+  assert.sameMembers(Array.from(new Set(directChildren)), expectedSpanNames);
+  // TODO:  All spans are closed??
```

Comment:
We should add this... something like:

```ts
const openSpans = instrumenter.startedSpans.filter(s => !s.endCalled);
assert.equals(openSpans.length, 0, `All spans should have been closed, but found ${openSpans.map(s => s.name)} open spans.`);
```
Or something. Note: you will need to close the root span after calling the callback...

Reply:
done both.